### PR TITLE
Distinguish between unit and integration tests

### PR DESCRIPTION
**Pull request by imscaradh**

### CODE REVIEW

File: src/service/recorder/influxGravityDataRecorder.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public start(measurement: string, tags: any = {}, transformer: TransformerFn) {
+        this.record = true;
+        this.mqttClient.onTopicMessage(this.topicDefinitions.gravity, (async (message: Buffer) => {
+            // FIXME: THIS is not accessible at callback execution time
+            // this.frameCount++;
+            // if (!this.isRecording() || this.frameCount % this.nthFrame != 0) {
+            //     return;
+            // }
```
Comment (curtys) [inline src/service/recorder/influxGravityDataRecorder.ts]: THIS is accessible and references correctly the recorder instance. When creating the anonymous function THIS is bound.

File: src/service/recorder/influxDataRecorder.ts [hunk 2]
```diff
@@ -1,41 +0,0 @@
-import { InfluxDB } from 'influx';
```
Comment (curtys) [inline src/service/recorder/influxDataRecorder.ts]: Why did you rename the file? The idea was that this recorder could be used to store any characteristic to Influx. Characteristic specific data transformation is supposed to be done by an appropriate Transformer function.

File: src/service/recorder/dataRecorder.ts [hunk 3]
```diff
@@ -1,39 +1,21 @@
...
-                sound: `${config.macThingy2}/Thingy Sound Service/Thingy Microphone Characteristic`,
-            },
-            thingy3: {
-                connected: `${config.macThingy3}/Connected`,
-                gravity: `${config.macThingy3}/Thingy Motion Service/Thingy Gravity Characteristic`,
-                acceleration: `${config.macThingy3}/Thingy Motion Service/Thingy Acceleration Characteristic`,
-                sound: `${config.macThingy3}/Thingy Sound Service/Thingy Microphone Characteristic`,
-            },
+    protected constructor(protected mqttClient: MQTTTopicClient, protected device: any) {
+        this.topicDefinitions = {
```
Comment (curtys) [inline src/service/recorder/dataRecorder.ts]: I think this would be best placed in the thingy lib file.

File: src/service/recorder/influxDataRecorder.ts (hunk 2 above)
Comment (imscaradh) [reply src/service/recorder/influxDataRecorder.ts]: Because this recorder type contains x,y,z fields and is therefore gravity specific, right?

File: src/service/recorder/influxGravityDataRecorder.ts (hunk 1 above)
Comment (imscaradh) [reply src/service/recorder/influxGravityDataRecorder.ts]: My tests gave opposite results, `this.isRecording()` was always false. It would be best to do a peer debugging session for this ;)

Comment (curtys) [reply src/service/recorder/influxGravityDataRecorder.ts]: Strange. "It works on my machine" ;) Maybe you didn't stop the recorder? `this.isRecording()` is only false after stopping. We can do a debugging session if you want.

File: src/service/recorder/influxDataRecorder.ts (hunk 2 above)
Comment (curtys) [reply src/service/recorder/influxDataRecorder.ts]: Yes because I was to lazy to implement it properly ;) Will fix
